server/src/main/java/io/crate/planner/optimizer/rule/MoveConstantJoinConditionsBeneathNestedLoop.java

```diff
@@ -102,7 +103,7 @@ public LogicalPlan apply(NestedLoopJoin nl,
                 newLhs,
                 newRhs,
                 AndOperator.join(nonConstantConditions),
-                nl.topMostLeftRelation()
+                getOnlyElement(rhs.baseTables())
```

Comment:
See https://github.com/crate/crate/blob/77f7fbac26c7eef2af44ba87445b259a02bd6d1d/server/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java#L173-L177

Reply:
Is it safe to assume the right side only has a single base table?

We build the join tree starting from left and don't have bushy joins, but what if there is a sub-select with a union or something like that?

Reply:
So the precondition to this optimization is that it is an inner-join which has to be an equi-join. An equi-join has the condition that at least one argument one argument must no contain multiple tables. Would it work to detect which side has only one argument and use that as the concrete relation ?

Reply:
In case when both sides have only one argument i guess it should be the smaller table, right ?

Reply:
Ok, the whole assumption was wrong. An inner-join does not have to be an equi-join necessarly.